BOLT 1: make remote_addr definition machine readable.

rustyrussell opened the pull request:
It had a blank line and invalid format for tools/extract-formats.py.  This meant that the python package omitted it!

And move the format information into the requirements section
(and fix spelling: `node_announement` -> `node_announcement`

Diff for extract-formats.py before and after:

```diff
--- /tmp/before	2022-05-17 10:47:01.583086352 +0930
+++ /tmp/after	2022-05-17 10:51:59.166850111 +0930
@@ -6,6 +6,8 @@
 msgdata,init,tlvs,init_tlvs,
 tlvtype,init_tlvs,networks,1
 tlvdata,init_tlvs,networks,chains,chain_hash,...
+tlvtype,init_tlvs,remote_addr,3
+tlvdata,init_tlvs,remote_addr,data,byte,...
 msgtype,error,17
 msgdata,error,channel_id,channel_id,
 msgdata,error,len,u16,
```

Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>

m-schmoock commented:
> Diff for extract-formats.py before and after:

It will name the structure just `data`
```
tlvtype,init_tlvs,remote_addr,3
tlvdata,init_tlvs,remote_addr,data,byte,...
 ```
 
 If thats okay, I can ack

t-bast commented on 01-messaging.md:
```diff
@@ -284,7 +283,9 @@ The sending node:
-  - SHOULD NOT set private addresses as `remote_addr`.
+  - if it sets `remote_addr`:
+     - MUST set it to a valid `address descriptor` (1 byte type and data) as described in [BOLT 7](07-routing-gossip.md#the-node_announcement-message).
+    - SHOULD NOT set private addresses as `remote_addr`.
```
nit: this isn't aligned with the other above

rustyrussell replied:
Yep, the latter was correct (2 spaces per indent), so I fixed the upper.

m-schmoock commented:
Seems like I still can't push into Rusty's branch. We need to wait for him...

rustyrussell commented:
> > Diff for extract-formats.py before and after:
> 
> It will name the structure just `data`
> 
> ```
> tlvtype,init_tlvs,remote_addr,3
> tlvdata,init_tlvs,remote_addr,data,byte,...
> ```
> 
> If thats okay, I can ack

Fixed, description, thanks! (I changed it at the last moment, after I generated that diff!)

t-bast reviewed (approved):
ACK https://github.com/lightning/bolts/pull/990/commits/f1c797df2966237244527c1c6343dbe9bc765342

vincenzopalazzo reviewed (approved):
ACK https://github.com/lightning/bolts/pull/990/commits/f1c797df2966237244527c1c6343dbe9bc765342

rustyrussell commented:
OK, applying under the "spelling and typos" rule.
